x-pack/plugins/security_solution/common/detection_engine/types.ts

```diff
@@ -8,3 +8,102 @@ import { AlertAction } from '../../../alerts/common';
 export type RuleAlertAction = Omit<AlertAction, 'actionTypeId'> & {
   action_type_id: string;
 };
+
+export type SearchTypes =
+  | string
+  | string[]
+  | number
+  | number[]
+  | boolean
+  | boolean[]
+  | object
+  | object[]
+  | undefined;
+
+export interface Explanation {
+  value: number;
+  description: string;
+  details: Explanation[];
+}
+
+export interface TotalValue {
+  value: number;
+  relation: string;
+}
+
+export interface BaseHit<T> {
+  _index: string;
+  _id: string;
+  _source: T;
+}
+
+export interface EqlSequence<T> {
+  join_keys: SearchTypes[];
+  events: Array<BaseHit<T>>;
+}
+
+export interface EqlSearchResponse<T> {
+  is_partial: boolean;
+  is_running: boolean;
+  took: number;
+  timed_out: boolean;
+  hits: {
+    total: TotalValue;
+    sequences?: Array<EqlSequence<T>>;
+    events?: Array<BaseHit<T>>;
+  };
+}
+
+export interface ShardsResponse {
+  total: number;
+  successful: number;
+  failed: number;
+  skipped: number;
+  failures?: ShardError[];
+}
+
+export interface ShardError {
+  shard: number;
+  index: string;
+  node: string;
+  reason: {
+    type: string;
+    reason: string;
+    index_uuid: string;
+    index: string;
+    caused_by: {
+      type: string;
+      reason: string;
+    };
+  };
+}
+
+export interface SearchResponse<T> {
```

Comment:
why are we duplicating the  `SearchResponse` from `import { SearchResponse } from 'elasticsearch';`

Reply:
```
export interface SearchResponse<T> {
    took: number;
    timed_out: boolean;
    _scroll_id?: string;
    _shards: ShardsResponse;
    hits: {
        total: number;
        max_score: number;
        hits: Array<{
            _index: string;
            _type: string;
            _id: string;
            _score: number;
            _source: T;
            _version?: number;
            _explanation?: Explanation;
            fields?: any;
            highlight?: any;
            inner_hits?: any;
            matched_queries?: string[];
            sort?: string[];
        }>;
    };
    aggregations?: any;
}
```

Reply:
I see what you're saying. I moved this back to where it was since it wasn't needed in the UI either way. But this could be something to 👁️ to figure out if we can swap instances of this with the one from `elasticsearch`.

Reply:
It might be the reasoning for this one, but for a few of these the reason why we cannot use:

```
import { SearchResponse } from 'elasticsearch';
```

Is because it's not compatible with the cluster interfaces given to us from the alerting plugin so we have to use these older structures.